lti_consumer/lti_consumer.py

```diff
@@ -756,14 +901,56 @@ def lti_launch_handler(self, request, suffix=''):  # pylint: disable=unused-argu
         Returns:
             webob.response: HTML LTI launch form
         """
-        lti_consumer = LtiConsumer(self)
-        lti_parameters = lti_consumer.get_signed_lti_parameters()
+        if self.lti_version == "lti_1p3":
+            return self._launch_lti_1p3(request)
+
+        return self._launch_lti_1p1()
+
+    @XBlock.handler
+    def lti_launch_callback(self, request, suffix=''):  # pylint: disable=unused-argument
+        """
+        XBlock handler for launching the LTI provider.
+
+        Returns:
+            webob.response: HTML LTI launch form
+        """
+        if self.lti_version != "lti_1p3":
+            return Response('', content_type='text/html')
+
+        lti_consumer = self._get_lti1p3_consumer()
         loader = ResourceLoader(__name__)
-        context = self._get_context_for_template()
-        context.update({'lti_parameters': lti_parameters})
-        template = loader.render_mako_template('/templates/html/lti_launch.html', context)
+        context = {}
+
+        context.update({
+            "preflight_response": dict(request.GET),
+            "launch_request": lti_consumer.generate_launch_request(
+                user_id=self.runtime.user_id,
```

Comment:
I'm just using user_id for now while we clear up the questions around `external_user_id`.

Reply:
This is waiting on the `external_user_id` XBlock runtime implementation.